contrib/valkey-go/valkey.go

```diff
@@ -0,0 +1,367 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package redis provides tracing functions for tracing the go-redis/redis package (https://github.com/go-redis/redis).
+// This package supports versions up to go-redis 6.15.
+package valkey
+
+import (
+	"context"
+	"math"
+	"net"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/valkey-io/valkey-go"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/telemetry"
+)
+
+const componentName = "valkey-go/valkey"
+
+func init() {
+	telemetry.LoadIntegration(componentName)
+	tracer.MarkIntegrationImported("github.com/valkey/valkey-go")
```

Comment:
After importing, we should expect that there is matching code in `ddtrace/tracer/option.go` to support marking the integration imported. For example, we have a `contribIntegrations` slice that should hold all integrations, and relevant tests in `ddtrace/tracer/option_test.go` should also be updated.